kivakit-configuration/src/main/java/com/telenav/kivakit/configuration/BaseComponent.java

```diff
@@ -3,204 +3,82 @@
- *     <li>{@link #require(Class)} - Gets the given configuration object or fails</li>
- *     <li>{@link #require(Class, InstanceIdentifier)} - Gets the configuration object for the given instance or fails</li>
- * </ul>
- *
- * <p><b>Object Lookup Registry</b></p>
- *
- * <p>
- * Access to this component's lookup {@link Registry} is provided and fulfills basic needs for object wiring:
- * </p>
- *
- * <ul>
- *     <li>{@link #registry()} - Retrieves the lookup {@link Registry} for this component</li>
- *     <li>{@link #lookup(Class)} - Lookup the registered object of the given type, if any</li>
- *     <li>{@link #lookup(Class, String)} - Find the given instance of the given class, if any</li>
- *     <li>{@link #lookup(Class, Enum)} - Find the given instance of the given class, if any</li>
- *     <li>{@link #lookup(Class, InstanceIdentifier)} - Find the given instance of the given class, if any</li>
- *     <li>{@link #register(Object)} - Register the given singleton object for lookup</li>
- *     <li>{@link #register(Object, String)} - Register the given instance of the given object for lookup</li>
- *     <li>{@link #register(Object, Enum)} - Register the given instance of the given object for lookup</li>
- *     <li>{@link #register(Object, InstanceIdentifier)} - Register the given instance of the given object for lookup</li>
- * </ul>
+ * well as settings registration and lookup (see {@link Settings}). For details, see {@link Component}.
  *
  * @author jonathanl (shibo)
+ * @see Component
+ * @see Registry
+ * @see Settings
  */
-public class BaseComponent extends BaseRepeater
+public class BaseComponent extends BaseRepeater implements Component
 {
     /**
      * @return Any registered object of the given type with the given instance identifier in the global lookup registry
      */
-    public <T> T lookup(final Class<T> type, final String instance)
-    {
-        return registry().find(type, instance);
-    }
-
-    /**
-     * @return Any registered object of the given type with the given instance identifier in the global lookup registry
-     */
+    @Override
     public <T> T lookup(final Class<T> type, final InstanceIdentifier instance)
     {
-        return registry().find(type, instance);
-    }
-
-    /**
-     * @return Any registered object of the given type with the given instance identifier in the global lookup registry
-     */
-    public <T> T lookup(final Class<T> type, final Enum<?> instance)
-    {
-        return registry().find(type, instance);
-    }
-
-    /**
-     * @return The object of the given type from the application's lookup {@link Registry}, if any.
-     */
-    public <T> T lookup(final Class<T> type)
-    {
-        return registry().find(type);
-    }
-
-    /**
-     * Registers the given singleton object in the global lookup registry
-     */
-    public <T> T register(final T object)
-    {
-        return registry().add(object);
+        return registry().lookup(type, instance);
     }
```

Comment:
I am thinking seems implementation could be moved into interface Component as default method. There is no dependent fields of BaseComponent.

Reply:
Yes, I agree. I did exactly what you're suggesting in the develop branch after posting this review. Sorry.